Src/VimCore/Interpreter_Interpreter.fs

```diff
@@ -1454,14 +1454,16 @@ type VimInterpreter
                 if System.IO.Directory.Exists(vimFolder) then
                     let latest =
                         System.IO.Directory.EnumerateDirectories vimFolder
-                        |> Seq.map (fun pathName -> System.IO.Path.GetFileName(pathName))
-                        |> Seq.filter (fun folder ->
-                            folder.StartsWith("vim", System.StringComparison.OrdinalIgnoreCase))
-                        |> Seq.sortByDescending (fun folder ->
-                            folder.Substring(3)
-                            |> Seq.takeWhile CharUtil.IsDigit
-                            |> System.String.Concat
-                            |> int)
+                        |> Seq.map (fun pathName ->
+                            let folder = System.IO.Path.GetFileName(pathName)
+                            let m = System.Text.RegularExpressions.Regex.Match(folder, "^vim(\d+)", System.Text.RegularExpressions.RegexOptions.IgnoreCase)
+                            if m.Success then
+                                (folder, m.Groups.[1].Value |> int)
+                            else
+                                ("", 0))
+                        |> Seq.filter (fun (folder, version) -> folder <> "" && version <> 0)
+                        |> Seq.sortByDescending (fun (_, version) -> version)
+                        |> Seq.map (fun (folder, _) -> folder)

```

Comment:
This is perfectly fine, but since you asked for a review, two small things come to mind.

1. In imperative languages it is common to use sentinels, i.e. values with special significance. For example, -1. In F# the pattern here might be to use `Some tuple` vs. `None` and then filter to the tuples.

2. For readability at a higher indent level, the lambda function could be factored out into a named function defined above and used once.

Reply:
I think I understand what you mean in 1 and totally understand 2. When I attempted the Some/None approach, I ended up with a nested Option, so later on, `latest` was `string option option` instead of `string option`.

Reply:
Updated it by extracting it and changing it to use `Seq.choose`. Looks quite a bit more F# like and reads a bit easier. Thanks for the feedback and instruction @ricksladkey!